WordPress/gutenberg
# Fix relative links in main package READMEs so that they work in npm

## Description
Follow up to #38605.

In that PR, I discovered when npm displays the README for the package, it interprets relative links incorrectly resulting in a broken URL. This is probably an npm bug, but I haven't figured out where to report it yet (one of these - https://github.com/orgs/npm/repositories?type=all).

This PR updates some of the relative links I found using find/replace, changing them to absolute links.

## Testing Instructions
To reproduce the bug, try the links in the first paragraph of the core-data package in npm - https://www.npmjs.com/package/@wordpress/core-data.

You can see they're broken but they work ok in github - https://github.com/WordPress/gutenberg/tree/trunk/packages/core-data

If you try an absolute link in npm, like the one in the second 'Installation' section for core-data, that seems to work fine - https://www.npmjs.com/package/@wordpress/core-data.

So it seems like changing to absolute URLs is the best thing to do right now.

## Types of changes
Docs
